benchmarks/common/benchmark_runner.hpp

```diff
@@ -325,19 +326,44 @@ struct BenchmarkRunner {
     
   private:
     static void run_benchmark(benchmark::State& state, const Options& options,  Gemm gemm_op) {
-      state.counters["runtime_ms"] = 0;
+      auto tflop = ((2.0 * options.m * options.n * options.k * options.l) * 1e-12);
+      auto giga_bytes_transferred = ((options.m * options.k) * sizeof(ElementA) +
+                                      (options.k * options.n) * sizeof(ElementB) + 
+                                  2 * (options.m * options.n) * sizeof(ElementC)) * 1e-9;
+      initialize_counters(state);
       for(auto _ : state) {
         GPU_Clock timer;
         timer.start();
         gemm_op.run(); 
         auto ms_elapsed = timer.milliseconds();
-        state.counters["runtime_ms"] +=  ms_elapsed;
+        update_counters(state, ms_elapsed, tflop, giga_bytes_transferred);
         state.SetIterationTime(ms_elapsed / 1000);
       }
-      state.counters["runtime_ms"] /= state.iterations();
-      state.counters["TFlops"] = ((2.0 * options.m * options.n * options.k * options.l) * 1e-12) / 
-                                  (state.counters["runtime_ms"] / 1000);
+      finalize_counters(state, tflop, giga_bytes_transferred);
     }
 
+    static void initialize_counters(benchmark::State& state) {
+      state.counters["avg_runtime_ms"] = 0;
+      state.counters["avg_tflops"] = 0;
+      state.counters["avg_throughput"] = 0;
+      state.counters["best_runtime_ms"] = std::numeric_limits<double>::max();
+    }
+
+    static void update_counters(benchmark::State& state, double ms_elapsed, double tflop, double giga_bytes_transferred) {
+      state.PauseTiming();
+      state.counters["avg_runtime_ms"] += ms_elapsed;
+      state.counters["avg_tflops"] += tflop / (ms_elapsed * 1000);
+      state.counters["avg_throughput"] += giga_bytes_transferred / (ms_elapsed / 1000);
+      state.counters["best_runtime_ms"] = std::min<double>(state.counters["best_runtime_ms"], ms_elapsed);
+      state.ResumeTiming();
+    }
+
+    static void finalize_counters(benchmark::State& state,  double tflop, double giga_bytes_transferred) {
+      state.counters["avg_runtime_ms"] /= state.iterations();
+      state.counters["avg_tflops"] /= state.iterations();
+      state.counters["avg_throughput"] /= state.iterations();
```

Comment:
Same for `avg_throughput`. Also should it be `avg_bandwidth` for consistency with  `peak_bandwidth`

Reply:
^ same as above.

> Also should it be avg_bandwidth for consistency with peak_bandwidth

I do not suppose so. Bandwidth refers to how fast the device can transfer the data, at that certain clock speed. peak bandwidth refers to the bandwidth value at the highest power setting (highest clock values).
Throughput refers to how much of that is actually achieved